tests/out/shapes.txt

```diff
@@ -1,1774 +1,1774 @@
-3153802,53.0370344,8.8223766,8
-3153802,53.0365116,8.8176798,9
-3153802,53.0326937,8.8395021,10
-3153802,53.0517842,8.8198571,11
-3153803,53.0509756,8.8560113,0
-3153803,53.0352591,8.8260138,1
-3153803,53.0338236,8.8429917,2
-3153803,53.0537609,8.8508126,3
-3153803,53.0554537,8.8442286,4
-3153803,53.0543613,8.8409939,5
-3153803,53.0525009,8.8400534,6
-3153803,53.0374124,8.821674,7
-3153803,53.032495,8.8398022,8
-3170153,53.0394778,8.8155034,0
-3170153,53.0286287,8.8066292,1
-3185196,53.0602695,8.7641142,0
-3185196,53.0615425,8.7741344,1
-3185196,53.0590256,8.7780123,2
-3185196,53.0609753,8.7848701,3
-3185196,53.056284,8.7839847,4
-3185196,53.0541823,8.7854235,5
-3185196,53.0538966,8.7915421,6
-3185196,53.056006,8.7953486,7
-3185196,53.0465167,8.8189275,8
-3185196,53.0440043,8.8241022,9
-3185196,53.0409061,8.8281429,10
-3336378,53.051805,8.8189593,0
-3336378,53.0821074,8.8135271,1
-3336379,53.0516787,8.8193566,0
-3336380,53.1734602,8.6464658,0
-3373529,53.1710163,8.6324922,0
-3373529,53.1853598,8.6169006,1
-3373529,53.1852427,8.5942087,2
-3373529,53.183712,8.5806046,3
-3373529,53.185488,8.5706177,4
-3373529,53.1889315,8.5573298,5
-3373529,53.1935036,8.5447688,6
-3373529,53.2048843,8.5158145,7
-3373530,53.1708252,8.6322796,0
-3373530,53.172662,8.6522585,1
-3373530,53.1714328,8.6712029,2
-3373530,53.1707061,8.6882194,3
-3373530,53.1596948,8.7048867,4
-3373530,53.1350051,8.7425975,5
-3373530,53.103367,8.7865402,6
-3373530,53.0604558,8.8889177,7
-3373530,53.0393023,8.9459948,8
-4261048,53.5350531,8.5997911,0
-4425378,53.0833147,8.8148343,0
-4425378,53.0877023,8.9356592,1
-4458994,53.0835459,8.8140657,0
-4593209,53.1769784,8.6847747,0
-4593209,53.1800358,8.7320639,1
-5346093,53.567878,8.5653882,0
-5346093,53.5693902,8.5604888,1
-5346093,53.5731997,8.5602197,2
-5346093,53.5811046,8.5478947,3
-5346093,53.5959072,8.5397568,4
-5513686,53.0865927,8.9403178,0
-5513686,53.0840909,8.8127024,1
-5626115,53.5350288,8.5995133,0
-5626115,53.5088309,8.6115355,1
-5754318,53.0821401,8.8105896,0
-6129984,53.194396,8.6149912,0
-6183796,53.1078054,8.7410882,0
-6183797,53.096256,8.7460892,0
-6251915,53.0719515,8.8030042,0
-6251916,53.0845593,8.8167823,0
-6251916,53.0862898,8.8181929,1
-6251919,53.0866887,8.8183064,0
-6251919,53.0843903,8.8166559,1
-6251919,53.0614534,8.7868959,2
-6253014,53.0847515,8.8509305,0
-6559182,53.1907488,8.6354251,0
-6559182,53.1906215,8.6311785,1
-6559182,53.188182,8.6293942,2
-6559182,53.1841665,8.6264094,3
-6559182,53.1781634,8.6247429,4
-6559182,53.1744102,8.623985,5
-6559182,53.1697553,8.6283579,6
-6559183,53.1752794,8.6242875,0
-6559183,53.178893,8.6249299,1
-6559183,53.1833566,8.6260986,2
-6559183,53.182946,8.6343984,3
-6559183,53.1858632,8.6376531,4
-6559183,53.1885375,8.6406411,5
-6560472,53.1947326,8.6150382,0
-6560472,53.1925607,8.6171911,1
-6560472,53.190696,8.6184409,2
-6560472,53.1861406,8.6191371,3
-6560472,53.1700892,8.6279496,4
-6648747,53.1214837,8.7526593,0
-6785438,53.1684836,8.6615385,0
+19914,8.7896481,53.0507001,0
```

Comment:
@mehdisadeghi the lat and lon are swapped comparing them with the previous output. If I have to guess I would say that the current one is wrong since you are using Germany's data which means lat around 50 and lon around 8

Reply:
@carocad oh wait let me see, too much refactoring today! 😆

Reply:
![image](https://user-images.githubusercontent.com/3101557/33563869-576cd954-d919-11e7-8a2d-8e53163c8794.png)

Yeah...